src/main/java/uk/ac/ebi/ampt2d/accession/service/MonotonicAccessionGenerator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.service;
+
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.ac.ebi.ampt2d.accession.serial.block.MonotonicRangePriorityQueue;
+import uk.ac.ebi.ampt2d.accession.serial.block.persistence.entities.ContiguousIdBlock;
+import uk.ac.ebi.ampt2d.accession.serial.block.persistence.repositories.ContiguousIdBlockRepository;
+import uk.ac.ebi.ampt2d.accession.service.exceptions.AccessionIsNotPending;
+import uk.ac.ebi.ampt2d.accession.utils.ExponentialBackOff;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Monotonic accession, generates blocks of ids on database with a blocking serialized transaction. Each block
```

Comment:
Is there something missing in the first sentence of this javadoc block?

Reply:
It sounds a bit weird indeed, maybe you wanted to say something like this?

> Generates blocks of monotonically increasing accessions on database with a blocking serialized transaction. Each block is linked to an applicationInstanceId that identifies which application instance created it, and a categoryId to identify the type of resource.